### PR TITLE
[Terra Clinical Label Value View] Link the definition list in the LabelValueView with a text provided by the consumer, that describes what the definition list is about

### PR DESCRIPTION
**Pull request by MadalinaStelea**
### Summary

**What was changed:**
- Created a new prop `topicTextId` on the LabelValueView component.
- Applied the aria-labelledby attribute on the definition list returned by the LabelValueView, to link it to a topic text defined by the consumer.
- Updated the component and tests examples for the new prop `topicTextId`.

**Why it was changed:**
The change was required for the accessibility purpose, so the user understand better each LabelValueView to which topic is associated.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9190 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/CHANGELOG.md]: Removed this line.

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/CHANGELOG.md]: Made the change.

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/LabelValueView.jsx]: It is the id of the text. I've updated the comment. Thank you!

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (supreethmr) [inline packages/terra-clinical-label-value-view/CHANGELOG.md:7]:
```suggestion

* Added
  * Added a new prop `topicTextId` on the LabelValueView component.
  
* Changed
  * Applied the `aria-labelledby` attribute on the definition list returned by the LabelValueView, to link it to a topic text defined by the consumer.
```

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (supreethmr) [inline packages/terra-clinical-label-value-view/src/LabelValueView.jsx:81]: Can we add Jest test for new prop to ensure aria-labelledBy is rendered only when `topicTextId` is provided with value.

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/CHANGELOG.md]: I made the change. Thanks!

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/LabelValueView.jsx:81]: I've added 4 new jest tests for both the new prop `topicTextId` and the prop I've added in my previous Jira 'isChildOfDescriptionList'. Thank you!

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (sdadn) [inline packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:22]:
```suggestion
LabelValueViews and the encapsulating description lists are often visually associated with a topic on the page. It is best practice to programmatically associate the LabelValueView with the ID of the element that labels it on the page by using one of the following methods:
```

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (sdadn) [inline packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:24]:
```suggestion
- If the `isChildOfDescriptionList` prop is set to `true` associate the description list with the label using the `aria-labelledby` attribute on the description list that points to the ID of the topic.
```

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/example/LabelValueViewTextTermAndDef.jsx
Comment (sdadn) [inline packages/terra-clinical-label-value-view/src/terra-dev-site/doc/example/LabelValueViewTextTermAndDef.jsx:13]: We can remove "Topic Text" as that's redundant.
```suggestion
    <h4 id="topicText">Vitals</h4>
```

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:14]: The implementation of the definition list was done after some discussions with the UX team, who decided that this is what we want for the LabelValueView, in order to make the association of a label-value pair and for the screen readers to clearly notify the user that the 2 elements are related, even if the textValue is not always a real definition. I agree that it is not always accurate, but we didn't find any better element to use and this is what UX considered being the best way the screen readers can notify the user of what the label and the value are.
Maybe @scottwilmarth can bring more arguments than me or can explain better the need of the definition list :) Or @sdadn - do you have any suggestion of how this can be done in a better way?

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/LabelValueView.jsx:87]: I made the suggested change. Thanks!

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/LabelValueView.jsx:27]: Made the change. Thank you!

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/example/LabelValueViewTextTermAndDef.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/example/LabelValueViewTextTermAndDef.jsx:13]: Done!

File: packages/terra-clinical-data-grid/CHANGELOG.md
Comment (sdadn) [inline packages/terra-clinical-data-grid/CHANGELOG.md:6]: This was already done by #879 , can you rebase with main so your PR is up to date?

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (brooksienoodlesoup) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx]: Hi @MadalinaStelea and @sdadn  -
I think it makes sense to remove the phrases "(a term)" and "(a definition)" from the previous description I helped craft.

Also, here is some additional explanation I just wrote that speaks to reasoning behind using LabelValueView to enhance access:

"The LabelValueView component supports easy content discovery for assistive technology (AT) users, who may search for and navigate by lists on the page.  Using LabelValueView also supports consistent and clear AT delivery of the component content by supporting separate review of label/value pairs in structured announcements to AT users."

Let me know if this works better.

Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx]: @brooksienoodlesoup - sorry for the late response. I had some issues with the wdio in the PR, that I've just fixed. I also removed "a term" and "a definition" from the text above.

File: packages/terra-clinical-data-grid/CHANGELOG.md
Comment (MadalinaStelea) [reply packages/terra-clinical-data-grid/CHANGELOG.md]: @sdadn sorry for the late response, I had some issues with wdio that I've just fixed. I did the rebase and the PR is up to date now.

File: packages/terra-clinical-label-value-view/src/LabelValueView.jsx
Comment (sycombs) [inline packages/terra-clinical-label-value-view/src/LabelValueView.jsx:27]: small nit, can we capitalize By since it is technically another word for consistency with other components: https://github.com/cerner/terra-core/blob/main/packages/terra-toolbar/src/Toolbar.jsx#L32 
```suggestion
  ariaLabelledBy: PropTypes.string,
```

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (sycombs) [inline packages/terra-clinical-label-value-view/CHANGELOG.md:5]: This should be updated with the new prop name.

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (sycombs) [inline packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:16]: Can we add two short code examples in this doc to illustrate the different usages?

**Review by sycombs — CHANGES_REQUESTED** [review]

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (sdadn) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:26]: Was this change committed?

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx:16]: I have already updated the examples in the previous Jira, to ilustrate the 2 different usage options of the LabelValueView - as a definition list or as individual term-definition pair. In the current PR, I have also set the aria-labelledby/ariaLabelledBy props for the existing examples.

File: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx]: Sure, I've made the change. Thank you for bringing it up, I missed a few places where I should have updated it.

File: packages/terra-clinical-label-value-view/CHANGELOG.md
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/CHANGELOG.md]: Right! Made the change. Thanks!

**Review by sdadn — APPROVED** [review]

**Review by sycombs — APPROVED** [review]

File: packages/terra-clinical-label-value-view/tests/jest/LabelValueView.test.jsx
Comment (MadalinaStelea) [reply packages/terra-clinical-label-value-view/tests/jest/LabelValueView.test.jsx]: Made this change as well, thank you!

**Comment by MadalinaStelea** [comment]
@brooksienoodlesoup and @supreethmr - could you please confirm you are ok with all the changes I've made after your suggestions and approve the PR if everything looks good to you?

**Review by supreethmr — APPROVED** [review]

**Pull request merged**
